### PR TITLE
fix(storage): list folders

### PR DESCRIPTION
**Pull request by grdsdev**
## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

`Bucket` and `FileObject` structs consider some optional fields as required, this breaks in cases where a _folder_ is returned from the `list` method, as folders have only the `name` as non-null.

## Additional context

Close https://github.com/supabase/supabase-swift/issues/479

### CODE REVIEW

File: Sources/Storage/FileObject.swift [hunk 1]
```diff
@@ -2,26 +2,24 @@ import Foundation
...
-  public var name: String
+  public var name: String?
   public var bucketId: String?
   public var owner: String?
-  public var id: String
-  public var updatedAt: Date
-  public var createdAt: Date
-  public var lastAccessedAt: Date
-  public var metadata: [String: AnyJSON]
-  public var buckets: Bucket?
```
Comment (grdsdev) [inline Sources/Storage/FileObject.swift:13]: from my tests this is never returned

File: Sources/Storage/FileObject.swift [hunk 2]
```diff
@@ -2,26 +2,24 @@ import Foundation
...
+    name: String? = nil,
     bucketId: String? = nil,
     owner: String? = nil,
-    id: String,
-    updatedAt: Date,
-    createdAt: Date,
-    lastAccessedAt: Date,
-    metadata: [String: AnyJSON],
-    buckets: Bucket? = nil
+    id: UUID,
```
Comment (Axxou) [inline Sources/Storage/FileObject.swift:18]: On my side, its only after I put the `id: UUID? = nil` that the fetch worked.

File: Sources/Storage/FileObject.swift [hunk 3]
```diff
@@ -2,26 +2,24 @@ import Foundation
...
+  public var name: String?
   public var bucketId: String?
   public var owner: String?
-  public var id: String
-  public var updatedAt: Date
-  public var createdAt: Date
-  public var lastAccessedAt: Date
-  public var metadata: [String: AnyJSON]
-  public var buckets: Bucket?
+  public var id: UUID
```
Comment (Axxou) [inline Sources/Storage/FileObject.swift:8]: To complete with my comment about optional id : `public var id: UUID?`

**Comment by grdsdev** [comment]
@fenos does this make sense from a backend perspective?

All but `name` field from `FileObject` struct are optional?

**Review by dshukertjr — APPROVED** [review]

**Pull request merged**